pkg/overlay/cache.go

```diff
@@ -127,3 +128,16 @@ func (o *Cache) Put(ctx context.Context, nodeID storj.NodeID, value pb.Node) err
 
 	return o.DB.Put(nodeID.Bytes(), data)
 }
+
+// ConnFailure implements the Transport Observer `ConnFailure` function
+func (o *Cache) ConnFailure(ctx context.Context, node *pb.Node, err error) {
+	// noop until we figure out what to do on ConnFailure
+}
+
+// ConnSuccess implements the Transport Observer `ConnSuccess` function
+func (o *Cache) ConnSuccess(ctx context.Context, node *pb.Node) {
+	err := o.Put(ctx, node.Id, *node)
```

Comment:
why do you call `cache.Put()` here if you are also calling it from `discovery.ConnectionSuccess()`?